runtime/v2/manager.go

```diff
@@ -604,14 +602,35 @@ func registerServices[T transaction.Tx](s hasServicesV1, app *App[T], registry *
 		err:               nil,
 	}
 
-	if err := s.RegisterServices(c); err != nil {
-		return fmt.Errorf("unable to register services: %w", err)
+	if services, ok := s.(hasServicesV1); ok {
+		if err := services.RegisterServices(c); err != nil {
+			return fmt.Errorf("unable to register services: %w", err)
+		}
+	} else {
+		// If module not implement RegisterServices, register msg & query handler.
+		if module, ok := s.(appmodulev2.HasMsgHandlers); ok {
+			module.RegisterMsgHandlers(app.msgRouterBuilder)
+		}
+
+		if module, ok := s.(appmodulev2.HasQueryHandlers); ok {
+			module.RegisterQueryHandlers(app.queryRouterBuilder)
+		}
 	}
 
 	if c.err != nil {
 		app.logger.Warn("error registering services", "error", c.err)
 	}
 
+	// TODO: query regist by RegisterQueryHandlers not in grpcQueryDecoders
```

Comment:
Do some hacky here cause `RegisterQueryHandlers` not add query decoder to `grpcQueryDecoders` so I have to add again.
Must be some way better but idk yet.